### DISCUSSION TITLE
How to use proxy_pass with this nginx proxy? Please help!

### DISCUSSION THREAD

**Discussion by Pryla** [nginx-proxy/nginx-proxy#1900] [Q&A]
Hello, I know questions like this almost never get answered but I am looking for a solution for many months now and I have not found a good solution. I would like to use [plausible analytics](https://plausible.io/). The script gets blocked by some adblockers so I need to bypass it with a proxy pass:

```
# Only needed if you cache the plausible script. Speeds things up.
# Note: to use the `proxy_cache` setup, you'll need to make sure the `/var/run/nginx-cache`
# directory exists (e.g. creating it in a build step with `mkdir -p /var/run/nginx-cache`)
proxy_cache_path /var/run/nginx-cache/jscache levels=1:2 keys_zone=jscache:100m inactive=30d  use_temp_path=off max_size=100m;

server {
    ...
    location = /js/script.js {
        # Change this if you use a different variant of the script
        proxy_pass https://plausible.io/js/plausible.js;

        # Tiny, negligible performance improvement. Very optional.
        proxy_buffering on;

        # Cache the script for 6 hours, as long as plausible.io returns a valid response
        proxy_cache jscache;
        proxy_cache_valid 200 6h;
        proxy_cache_use_stale updating error timeout invalid_header http_500;

        # Optional. Adds a header to tell if you got a cache hit or miss
        add_header X-Cache $upstream_cache_status;
    }

    location = /api/event {
        proxy_pass https://plausible.io/api/event;
        proxy_buffering on;
        proxy_http_version 1.1;

        proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host  $host;
    }
}
```
Source: [https://plausible.io/docs/proxy/guides/nginx](https://plausible.io/docs/proxy/guides/nginx)

Do I really need to another entire nginx server docker container in order to write my own location or is there another way?

I am really looking forward to have an answer. I think this question should be easily answered by someone who has some experience.

THANK YOU!

**Comment by buchdag** [comment]
I might be mistaken but it looks to me like you might be able to do this with [per-virtual host location configuration](https://github.com/nginx-proxy/nginx-proxy#per-virtual_host-location-configuration).

**Reply by Pryla** [reply]
Thank you very much for answering but unfornately this is not working because with this configuration I am already inside the main `location` part (behind the opening curly bracket). What I need is to say: `location = /js/script.js {...`

Do you have any other idea?

I probably could just add another nginx container and point this one to the reverse proxy but I am concerned about performance and ease of use.

**Reply by buchdag** [reply]
> Do you have any other idea?

[Per virtual host configuration.](https://github.com/nginx-proxy/nginx-proxy#per-virtual_host)

This will add your location under the corresponding `server` directive.

https://github.com/nginx-proxy/nginx-proxy/blob/fb8ddfd08c0f6cb53e583ea22ff8be06f69c50dc/nginx.tmpl#L427-L428
